test_server.py

```diff
@@ -0,0 +1,42 @@
+import server
+import socket
+import pytest
+from multiprocessing import Process
+
+
+@pytest.yield_fixture()
+def server_setup():
+    process = Process(target=server.run_server)
+    process.daemon = True
+    process.start()
+    yield process
+
+
+@pytest.fixture()
+def client_setup():
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(server.addr)
+    return client
+
+
+def test_response_ok(server_setup):
+    response_body = '<html><body><h1></h1>'\
+        '<p>All good here, captain.</p></body></html>'
+    assert response_body in server.response_ok()
```

Comment:
This test is a bit odd.  It would only work for one okay response, and it doesn't test any of the important parts of what makes a bytestring a well-formed HTTP response.